File: Low Level Design (LLD)/Problems/06. Splitwise/Java (with Springboot)/splitwise/src/main/java/com/splitwise/splitwise/model/Group.java
package com.splitwise.splitwise.model;

import java.util.*;

public class Group {
    private String groupId;
    private String name;
    private List<User> members = new ArrayList<>();
    private List<Expense> expenses = new ArrayList<>();

    public Group() {}
    public Group(String groupId, String name) {
        this.groupId = groupId;
        this.name = name;
    }

    public String getGroupId() { return groupId; }
    public String getName() { return name; }
    public List<User> getMembers() { return members; }
    public List<Expense> getExpenses() { return expenses; }

    public void addMember(User user) { members.add(user); }
    public void addExpense(Expense expense) { expenses.add(expense); }
}

File: Low Level Design (LLD)/Problems/06. Splitwise/Cpp/model/src/Group.cpp
#include "../include/Group.h"

Group::Group(const std::string& groupId, const std::string& name)
    : groupId(groupId), name(name) {}

const std::string& Group::getGroupId() const {
    return groupId;
}

const std::string& Group::getName() const {
    return name;
}

const std::vector<std::shared_ptr<User>>& Group::getMembers() const {
    return members;
}

const std::vector<std::shared_ptr<Expense>>& Group::getExpenses() const {
    return expenses;
}

void Group::addMember(const std::shared_ptr<User>& user) {
    members.push_back(user);
}

void Group::addExpense(const std::shared_ptr<Expense>& expense) {
    expenses.push_back(expense);
}
